File: web/src/jsMain/kotlin/com.musafira2z.store.web/ui/components/CartBody.kt
package com.musafira2z.store.web.ui.components

import androidx.compose.runtime.Composable
import com.musafira2z.store.fragment.CheckoutDetailsFragment
import com.musafira2z.store.web.ui.utils.toClasses
import com.musafira2z.store.web.ui.utils.toFormatPrice
import org.jetbrains.compose.web.dom.*

@Composable
fun CartBody(
    cart: CheckoutDetailsFragment,
    onClose: () -> Unit,
    onIncrement: (String) -> Unit,
    onDecrement: (String, Int) -> Unit,
    removeLine: (String) -> Unit,
    content: @Composable () -> Unit
) {
    Div(attrs = {
        toClasses("flex flex-col justify-end h-full overflow-hidden px-3")
    }) {
        Div {
            Div(attrs = {
                toClasses("bg-slate-100 p-2 flex justify-between")
            }) {
                Div(attrs = {
                    toClasses("flex items-center")
                }) {
                    IcShoppingCart()
                    H4(attrs = { classes("ml-2", "font-bold") }) {
                        Text("${cart.lines.size} Items")
                    }
                }
                Button(attrs = {
                    toClasses("bg-red-500 text-slate-50 rounded-lg text-6xl")
                    onClick {
                        onClose()
                    }
                }) {
                    Img(src = "icons/rx/cross-2.svg", attrs = {
                        attr("width", "40")
                        attr("height", "40")
                    })
                }
            }

            Div(attrs = { classes("h-140", "overflow-auto") }) {
                cart.lines.forEach { line ->
                    CartItem(
                        line = line,
                        onIncrement = {
                            onIncrement(line.checkoutLineDetailsFragment.variant.id)
                        },
                        onDecrement = {
                            val qty = line.checkoutLineDetailsFragment.quantity
                            if (qty > 1) {
                                onDecrement(
                                    line.checkoutLineDetailsFragment.variant.id,
                                    qty - 1
                                )
                            } else {
                                removeLine(line.checkoutLineDetailsFragment.id)
                            }
                        },
                        removeLine = removeLine
                    )
                }
            }
            Hr(attrs = { classes("h-4", "mt-5") })
        }
        Div(attrs = { classes("pb-6") }) {
            CartSummary(
                subTotal = cart.subtotalPrice.net.priceFragment.toFormatPrice(),
                discount = cart.discount?.priceFragment?.let {
                    if (it.amount > 0) {
                        it.toFormatPrice()
                    } else null
                },
                total = cart.totalPrice.gross.priceFragment.toFormatPrice()
            ) {
                Div(attrs = {
                    toClasses("grid grid-cols-6 text-slate-50 font-bold mt-2 bg-green-500 hover:bg-green-600 p-1 rounded-md")
                }) {
                    content()
                }
            }
        }
    }
}

File: src/Components/Cart/CartBody.js
import React from 'react';
import { RxCross2 } from 'react-icons/rx';
import AddToCartCard from './AddToCartCard';
import ProductCalculation from './ProductCalculation';

const CartBody = ({ children, isOpen, setIsOpen, checkoutData }) => {

    return (
        <div className={`flex flex-col  justify-between `}>


            <div className=' flex !justify-between px-5 py-5'>
                <div className='flex  items-center text-black'>
                    <img src="/favicon.ico" alt="" className='h-5' />  <h4 className='ml-2 font-bold'>  {checkoutData?.lines.length || "00"}Items</h4>
                </div>
                <button
                    onClick={() => setIsOpen(!isOpen)}
                    className={`text-gray-800 flex justify-center items-center border px-3 border-dashed border-gray-800 border-1 rounded-lg `}>
                    Close <RxCross2 size={15} />
                </button>

            </div>

            <div className=' h-[50vh] overflow-y-auto'>
                {
                    checkoutData?.lines?.map((data, i) => (
                        <AddToCartCard key={i} data={data} />

                    ))

                }

            </div>




            <div className='  border-t '>
                <div className='px-3'>
                    <ProductCalculation checkoutData={checkoutData}>

                        <div className='   text-slate-50 font-bold mt-2  p-1 rounded-md'>

                            {children}

                        </div>

                    </ProductCalculation>
                </div>
            </div>
        </div>
    );
};

export default CartBody;